### PR TITLE
second batch

### PR DESCRIPTION
**Pull request by croqueGrec09**

### CODE REVIEW

File: grails-app/controllers/de/laser/IssueEntitlementController.groovy [hunk 1]
```diff
@@ -57,7 +57,7 @@ class IssueEntitlementController  {
       result.editable = result.issueEntitlementInstance.subscription.isEditableBy(result.user)
 
       // Get usage statistics
-      def title_id = result.issueEntitlementInstance.tipp.title?.id
+      def title_id = result.issueEntitlementInstance.tipp.name
```
Comment (Moetez16) [inline grails-app/controllers/de/laser/IssueEntitlementController.groovy:60]: title_id != tipp.name

File: grails-app/services/com/k_int/kbplus/ExportService.groovy [hunk 2]
```diff
@@ -1185,7 +1181,7 @@ class ExportService {
 				//localprice_currency
 				row.add(entitlement.priceItem.localCurrency ? entitlement.priceItem.localCurrency.value : ' ')
 				//price_date
-				row.add(entitlement.priceItem.priceDate ? formatter.format(entitlement.priceItem.priceDate) : ' ')
+				row.add(entitlement.priceItem.startDate ? formatter.format(entitlement.priceItem.startDate) : ' ')
```
Comment (Moetez16) [inline grails-app/services/com/k_int/kbplus/ExportService.groovy:1184]: priceDate != startDate

Comment (croqueGrec09) [reply grails-app/services/com/k_int/kbplus/ExportService.groovy:1184]: In first batch, priceDate has been renamed in startDate to get congruence with WeKB

File: grails-app/controllers/de/laser/IssueEntitlementController.groovy (hunk 1 above)
Comment (croqueGrec09) [reply grails-app/controllers/de/laser/IssueEntitlementController.groovy]: Fixed in 3954815

File: grails-app/views/survey/surveyTitles.gsp [hunk 3]
```diff
@@ -194,10 +194,10 @@
                                     field="localCurrency" owner="${ie.priceItem}" overwriteEditable="${false}"
                                     config="Currency"/>
                         <%--<g:formatNumber number="${ie.priceItem.localPrice}" type="currency" currencyCode="${ie.priceItem.localCurrency.value}" currencySymbol="${ie.priceItem.listCurrency.value}"/>--%>
-                            (<g:message code="tipp.priceDate"/> <semui:xEditable field="priceDate"
-                                                                                 type="date"
-                                                                                 owner="${ie.priceItem}"
-                                                                                 overwriteEditable="${false}"/>
+                            (<g:message code="tipp.priceStartDate"/> <semui:xEditable field="priceDate"
```
Comment (Moetez16) [inline grails-app/views/survey/surveyTitles.gsp:197]: priceDate != startDate

Comment (croqueGrec09) [reply grails-app/views/survey/surveyTitles.gsp:197]: see above

File: grails-app/views/issueEntitlement/show.gsp [hunk 4]
```diff
@@ -131,9 +131,9 @@
                                                                               overwriteEditable="false"/> <semui:xEditableRefData
                                 field="localCurrency" owner="${issueEntitlementInstance.priceItem}" config="Currency"
                                 overwriteEditable="false"/>
-                        (<g:message code="tipp.priceDate"/> <semui:xEditable field="priceDate" type="date"
-                                                                             owner="${issueEntitlementInstance.priceItem}"
-                                                                             overwriteEditable="false"/>)
+                        (<g:message code="tipp.priceStartDate"/> <semui:xEditable field="priceDate" type="date"
+                                                                                  owner="${issueEntitlementInstance.priceItem}"
```
Comment (Moetez16) [inline grails-app/views/issueEntitlement/show.gsp:135]: priceDate != startDate

File: grails-app/taglib/de/laser/SemanticUiIconTagLib.groovy [hunk 5]
```diff
@@ -114,6 +114,7 @@ class SemanticUiIconTagLib {
                 dc = message(code: 'spotlight.databasetitle')
                 icon = 'database'
                 break
+            case 'Book':
```
Comment (Moetez16) [inline grails-app/taglib/de/laser/SemanticUiIconTagLib.groovy:117]: Doppelt case 'Book'

File: grails-app/services/de/laser/EscapeService.groovy [hunk 6]
```diff
@@ -34,7 +34,7 @@ class EscapeService {
     }
 
     BigDecimal parseFinancialValue(String input) {
-        String uniformedThousandSeparator = input.replaceAll('(.|,)(\\d{3})', '$2')
+        String uniformedThousandSeparator = input.replaceAll('(.|,|\')(\\d{3})', '$2')
```
Comment (Moetez16) [inline grails-app/services/de/laser/EscapeService.groovy:37]: Hast du das getestet ?

Comment (croqueGrec09) [reply grails-app/services/de/laser/EscapeService.groovy:37]: Der Sync nutzt das.

File: grails-app/views/issueEntitlement/show.gsp (hunk 4 above)
Comment (croqueGrec09) [reply grails-app/views/issueEntitlement/show.gsp:138]: fixed in 0701abf

File: grails-app/taglib/de/laser/SemanticUiIconTagLib.groovy (hunk 5 above)
Comment (croqueGrec09) [reply grails-app/taglib/de/laser/SemanticUiIconTagLib.groovy:117]: fixed in 0701abf

File: grails-app/views/survey/surveyTitles.gsp (hunk 3 above)
Comment (croqueGrec09) [reply grails-app/views/survey/surveyTitles.gsp]: fixed in 3c48c2b

File: grails-app/services/com/k_int/kbplus/GlobalSourceSyncService.groovy [hunk 7]
```diff
@@ -242,200 +232,272 @@ class GlobalSourceSyncService extends AbstractLockableService {
...
+                        priceItems: [],
+                        hostPlatformURL: tipp.url ?: null,
+                        identifiers: [],
+                        //id: tipp.'@id'.text(),
+                        uuid: tipp.uuid,
+                        accessStartDate : tipp.accessStartDate ? DateUtils.parseDateGeneric(tipp.accessStartDate) : null,
+                        accessEndDate   : tipp.accessEndDate ? DateUtils.parseDateGeneric(tipp.accessEndDate) : null,
+                        medium: tipp.medium
+                ]
+                if(tipp.titleType == 'JournalInstance') {
```
Comment (Moetez16) [inline grails-app/services/com/k_int/kbplus/GlobalSourceSyncService.groovy:390]: tipp.titleType == 'JournalInstance' != tipp.titleType == 'Journal'

Comment (croqueGrec09) [reply grails-app/services/com/k_int/kbplus/GlobalSourceSyncService.groovy]: fixed in efad8c4

**Review by Moetez16 — APPROVED** [review]

**Pull request merged**